src/api/modules/osu.ts

```diff
@@ -15,52 +15,76 @@ import {
   mockedAccountBalance,
   mockedDegrees,
 } from '../../mocks/osu';
-import { OSU_API_BASE_URL, OSU_API_CACHE_SEC } from '../../constants';
+import {
+  OSU_API_BASE_URL,
+  OSU_API_CACHE_SEC,
+  OSU_ERROR_SEC_THRESH,
+  OSU_ERROR_OCCUR_THRESH,
+} from '../../constants';
+import { cacheFailureOrPing } from './notifications';
 
 const STUDENT_BASE_URL: string = `${OSU_API_BASE_URL}/v1/students`;
 const PERSON_BASE_URL: string = `${OSU_API_BASE_URL}/v1/persons`;
 
-const getJson = async (url: string) => {
+const getJson = async (url: string, exceptionKey: string) => {
   // TODO: Can we cache this for a period of time and reuse the token reliably?
   const bearerToken = await getToken();
 
   // * Cached API call will not use the token until the time it's called
   // * after the cache had expired.
-  const response = await cache.get(
-    url,
-    {
-      headers: { Authorization: `Bearer ${bearerToken}` },
-      json: true,
-    },
-    true,
-    {
-      key: url,
-      ttlSeconds: OSU_API_CACHE_SEC,
-    },
-    [502],
-  );
-  return response;
+  try {
+    const response = await cache.get(
+      url,
+      {
+        headers: { Authorization: `Bearer ${bearerToken}` },
+        json: true,
+      },
+      true,
+      {
+        key: url,
+        ttlSeconds: OSU_API_CACHE_SEC,
+      },
+      [502],
+    );
+    return response;
+  } catch (err) {
+    const config = {
```

Comment:
probably just move the config object directly into the `await` line rather than declare and pass a variable.